### DISCUSSION TITLE
Can I provide a deploy key or otherwise make Yarn 3 resolve a package from a private GitHub repo?

### DISCUSSION THREAD

**Discussion by qcn** [renovatebot/renovate#15077] [Request Help]
Repro repository used as example: https://github.com/qcn-test-organization/main-repo
(https://github.com/qcn-test-organization/bunnies-are-really-great-qc also exists, but is private)

I have a project that installs a package from a separate private GitHub repository, using ssh:

```
"bunnies-are-really-great-qc": "ssh://git@github.com:qcn-test-organization/bunnies-are-really-great-qc.git",
```

We need to use the ssh format for our development workflows.

Using yarn v3, when GitHub-hosted Renovate [tries to update Yarn itself](https://github.com/qcn-test-organization/main-repo/pull/5#issuecomment-1047068421) on this repo, it fails because it's unable to update the `yarn.lock` file - it doesn't have permission to read the package. `ignoreDeps` doesn't make a difference, because it's not trying to update the dependency itself - it's running `yarn` to update the lockfile for the actual Yarn update. Is there a way to pass a [GitHub deploy key](https://docs.github.com/en/developers/overview/managing-deploy-keys) to Renovate to run that command?

**Comment by rarkins** [comment]
There's currently an open PR for specifying an sshKey via config

**Reply by qcn** [reply]
Sounds like my timing is perfect! Thank you :)
